src/search_datastructures.hpp

```diff
@@ -372,49 +375,79 @@ class GlobalDataHolder
+
+    using TRedQrySeqsRev   =
      seqan3::detail::lazy_conditional_t<c_transAlph == c_redAlph,
                         decltype(seqan3::views::type_reduce(std::declval<TTransQrySeqs &>())),   // no-op view
                         seqan3::detail::lazy_conditional_t<c_transAlph != AlphabetEnum::AMINO_ACID,
-                                        seqan3::detail::lazy<TRedNuclAlphModString, TTransQrySeqs, TRedAlph>,
-                                        seqan3::detail::lazy<TRedAlphModString, TTransQrySeqs, TRedAlph> > >;
+                                        seqan3::detail::lazy<TRedNuclAlphModString, TTransQrySeqs, TRedAlphRev>,
+                                        seqan3::detail::lazy<TRedAlphModString, TTransQrySeqs, TRedAlphRev> > >;
 
-    using TRedSbjSeqs   =
+    using TRedSbjSeqsFwd   =
      seqan3::detail::lazy_conditional_t<c_transAlph == c_redAlph,
                         decltype(seqan3::views::type_reduce(std::declval<TTransSbjSeqs &>())),   // no-op view
                         seqan3::detail::lazy_conditional_t<c_transAlph != AlphabetEnum::AMINO_ACID,
-                                        seqan3::detail::lazy<TRedNuclAlphModString, TTransSbjSeqs, TRedAlph>,
-                                        seqan3::detail::lazy<TRedAlphModString, TTransSbjSeqs, TRedAlph> > >;
+                                        seqan3::detail::lazy<TRedNuclAlphModString, TTransSbjSeqs, TRedAlphFwd>,
+                                        seqan3::detail::lazy<TRedAlphModString, TTransSbjSeqs, TRedAlphFwd> > >;
+
+    using TRedSbjSeqsRev   =
+     seqan3::detail::lazy_conditional_t<c_transAlph == c_redAlph,
+                        decltype(seqan3::views::type_reduce(std::declval<TTransSbjSeqs &>())),   // no-op view
+                        seqan3::detail::lazy_conditional_t<c_transAlph != AlphabetEnum::AMINO_ACID,
+                                        seqan3::detail::lazy<TRedNuclAlphModString, TTransSbjSeqs, TRedAlphRev>,
+                                        seqan3::detail::lazy<TRedAlphModString, TTransSbjSeqs, TRedAlphRev> > >;
 
     /* sequence ID strings */
     using TIds          = TCDStringSet<std::string>;
     using TQryIds       = std::vector<std::string>;
     using TSubjIds      = TIds;
 
     /* index */
-    using TIndexFile    = index_file<c_dbIndexType, c_origSbjAlph, c_redAlph>;
-    using TIndex        = typename TIndexFile::TIndex;
-    using TIndexCursor  = typename TIndex::cursor_type;
+    using TIndexFileFwd    = index_file<c_dbIndexType, c_origSbjAlph, c_redAlph, true>;
+    using TIndexFileRev    = index_file<c_dbIndexType, c_origSbjAlph, c_redAlph, false>;
+    using TIndexFwd        = typename TIndexFileFwd::TIndex;
+    using TIndexRev        = typename TIndexFileRev::TIndex;
+    using TIndexCursorFwd  = typename TIndexFwd::cursor_type;
+    using TIndexCursorRev  = typename TIndexRev::cursor_type;
```

Comment:
You could make the BS-only types be `void` by default to state more clearly that they are only used in that specific mode.